extensions/interactions/ItemSelectionInput/directives/ItemSelectionInputValidationService.js

```diff
@@ -144,11 +144,32 @@ oppia.factory('ItemSelectionInputValidationService', [
               });
             });
           });
-          areAllChoicesCovered = handledAnswers.every(function(handledAnswer) {
-            return handledAnswer;
+        } else {
+          answerGroups.forEach(function(answerGroup, answerIndex) {
```

Comment:
Why does this check only apply for `maxAllowedCount > 1`? Does it not apply generally?

Reply:
Sorry I missed that (I put it inside both, the if and the else condition. I will add it right now.

Reply:
In general, we should not be duplicating code. Why not run it as a separate check, outside both clauses? This seems like a separate check altogether from `areAllChoicesCovered`.

Reply:
Yeah @seanlip , I agree, I will do it. I should have done this, it will make #6418 easier.
Thanks!